### ISSUE TITLE
Rule proposal: `prefer-structured-clone`

### ISSUE THREAD

**Issue by B2o5T** [sindresorhus/eslint-plugin-unicorn#2091]
### Description

Available in node starting from v17

### Fail

```js
JSON.parse(JSON.stringify(foo))
```

```js
_.cloneDeep(foo)
```

### Pass

```js
structuredClone(foo)
```

**Comment by B2o5T** [comment]
Duplicate https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1640
